### ISSUE TITLE
Suggestion: Use node-xcode to dynamically add PSPDFKit framework

### ISSUE THREAD

**Issue by DavidStrausz** [PSPDFKit/Cordova-iOS#33]
I stumbled across this project and while going through your readme I thought you might wanna look into [cordova-node-code](https://github.com/apache/cordova-node-xcode). It offers a way to parse and modify ios-projects (the project.pbxproj) in javascript, cordova itself uses it under the hood. You could use it to add the PSPDFKit framework dynamically via a cordova hook which runs "after_platform_add".

**Comment by RDalziel** [comment]
+1 On this - i'm really struggling to get the library working on automated build. So many additional steps which there's a tonne of conflicting information on the net for.

@DavidStrausz Have you managed to get the latest PSPDFKit deployed as an embedded framework without manual intervention/

**Comment by DavidStrausz** [comment]
@RDalziel I'm not working with PSPDFKit specifically at the moment I just thought node-xcode would be a good fit for that use case. But I recently managed to to add a today-widget to a cordova-ios project dynamically and app-extensions are also embedded binaries. You can find my project [here](https://github.com/DavidStrausz/cordova-plugin-today-widget), maybe some things in there help you working with node-xcode (it's not very well documented unfortunately).

**Comment by RDalziel** [comment]
There may be a convenient mix of xcode_node & the latest embed="true" support in Cordova 7 - I'm still finding my way through Xcode project files but it seems as though "custom='true' embed='true'" gets you roughly 90% of the way there in this case. However cordova doesn't seem to handle the "LD_RUNPATH_SEARCH_PATHS" properly which results in library load errors on launch. 

Seems a little intricate to update this LD_RUNPATH_SEARCH_PATHS compiler option though - i'm experiencing inconsistent behaviour when i try to set it to   "\"$(inherited) @executable_path/Frameworks\"" - it doesn't seem to pick that up but has no issue picking up "\"$(inherited) @loader_path/Frameworks\""

**Comment by DavidStrausz** [comment]
I'd say the best way to debug here would be to create a working version only in XCode, backup the project.pbxproj file, delete the platform and readd it, then add the framework using cordova (custom=true, embed=true) and fix out the differences using node-xcode with a hook that runs after_plugin_add.

**Comment by RDalziel** [comment]
Cheers @DavidStrausz unfortunately i don't have a Mac available for this level of testing - i'm restrained to to CI Builds at the moment!

**Comment by DavidStrausz** [comment]
That sounds like terrible for debugging! I set up a VM running macOS for things like that 👍

**Comment by RDalziel** [comment]
The life on Windows wrestling with MAC...:D

**Comment by twinj** [comment]
You could use coca pods plugin. The Ionic 2 Demo shows how to get PSPDFKit installed automatically and also sets up entitlements.

**Comment by RDalziel** [comment]
So, embarrassingly, it took me about 3 days to figure this out. But there was a typo in my setup

I had 
<framework src="PSPDFKIT.framework"  custom="true" embed="true"/>

But it should have been 
<framework src="PSPDFKit.framework"  custom="true" embed="true"/>

A really silly mistake, however the reason it's difficult to find is there will be no errors in the entire build process and it will appear like everything is ok. However when you launch on device it will fail to find PSPDFKit due to the case difference.

It's worth noting that embed="true" works without issue on the latest version of Cordova and it is the easiest way to embed the PSPDFKit.Framework in the project.

**Comment by DavidStrausz** [comment]
@RDalziel Glad you could make it work! So as I understand no additional scripting is needed to make it work after plugin install? If so I'll close my issue.

**Comment by RDalziel** [comment]
There is additional scripting required but not on the embedded framework end. I know you're not focussing on this end but for others - the extra scripting is around removing the unused architectures to avoid issues publishing to Testflight/App Store

**Comment by DavidStrausz** [comment]
Ah! You mean the Run Script Phase! You can also use node-xcode to add additional build phases!

**Comment by RDalziel** [comment]
Indeed! PSPDF Supply a shell script which will strip archtecture in their source  - so i've used node-xcode to plug that in after the embedding.

Part of the difficulty is PSPDFKit not having their library in Github, so i've had to pull it down and have a persistent plugins/com/pspdfkit/cordovaplugin folder in my own repo which contains their library and architecture trimming script

**Comment by DavidStrausz** [comment]
You should defenitely submit a pull request after you're done with your local version 👍
